tools/protein_tools.py

```diff
@@ -0,0 +1,315 @@
+    Must contain start-codon and one of the stop-codons.
+    - protein (str): protein sequence translated from mRNA.
+    Must start with "M" and ends with "*" (stop-codon).
+
+    Note: is_protein(seq) doesn't see "*", but it's used in the other part of function.
+
+    Return:
+    - str, if mRNA without mutations return "Protein without mutations."
+    If there are mutations in protein, returns aminoacid(s) and their position(s)
+
+    Examples:
+    - "AUGGUAGGGAAAUUUUGA", "MVGKF*" ->  "Protein without mutations."
+    - "AUGGUAGGGAAAUUUUGA", "MGGVF*" -> "Mutations:G2, V4."
+    - "AUGGUAGGGAAAUUUUGA", "MGGKF" –> "ValueError: Stop (*) is absent"
+    - "AUGGUAGGGAAAUUUUGA", "GGKF*" –> "ValueError: Start (M) is absent"
+    - "AUGAAAAAAUGA", "MK*" -> "ValueError: Different length of translated protein and protein"
+    """
+    correct_protein = translate_rna(rna)
+    bank_of_mutations = []
+
+    if is_protein(protein[:-1]) is not True:
+        raise ValueError("Invalid protein sequence")
+    if is_rna(rna) is not True:
+        raise ValueError("Invalid RNA sequence")
+    if protein[-1] != "*":
+        raise ValueError("Stop (*) is absent")
+    if protein[0] != "M":
+        raise ValueError("Start (M) is absent")
+    if len(protein) != len(rna) / 3:
+        raise ValueError("Different length of translated protein and protein")
+
+    for i in range(len(correct_protein)):
+        if correct_protein[i] != protein[i]:
+            bank_of_mutations.append(f"{protein[i]}{i + 1}")
+
+    if len(bank_of_mutations) == 0:
+        return "Protein without mutations."
+    else:
+        return "Mutations: " + ", ".join(bank_of_mutations) + "."
+
+
+def run_protein_tools(*args: str):
+    """
+    Function containing methods for protein analysis.
+
+    Takes arbitrary number of arguments with protein sequencies
+    and the name of the procedure to be performed (always the last
+    argument). Returns the result of the procedure as string, tuple
+    or dictionary if one sequnce is submitted or list if several.
+
+    Note: if procedure 'check_mutations' is used then input must
+    contain only three arguments: RNA sequence, protein sequence
+    and the name of procedure itself.
+    """
+    *seqs, procedure = args
+    results = []
+    operations = {
+        "compute_molecular_weight": compute_molecular_weight,
+        "compute_length": compute_length,
+        "compute_hydrophobicity": compute_hydrophobicity,
+        "count_amino_acids": count_amino_acids,
+        "protein_to_dna": protein_to_dna
+
+    }
+    if procedure == "check_mutations":
+        results.append(check_mutations(seqs[0], seqs[1]))
+    else:
+        for seq in seqs:
+            if is_protein(seq) is not True:
+                raise ValueError("Invalid protein sequence")
+            if procedure not in operations:
+                raise ValueError("Wrong procedure name")
+            else:
+                results.append(operations[procedure](seq))
```

Comment:
Так было бы:
```suggestion
                results.append(operations[operation](seq))
```
Что покрасивше имхо